browser/ui/views/toolbar/brave_vpn_button.cc

```diff
@@ -50,11 +52,63 @@ class BraveVPNButtonHighlightPathGenerator
   }
 };
 
+class VPNButtonMenuModel : public ui::SimpleMenuModel,
+                           public ui::SimpleMenuModel::Delegate,
+                           public BraveVPNServiceObserver {
+ public:
+  explicit VPNButtonMenuModel(Browser* browser)
+      : SimpleMenuModel(this),
+        browser_(browser),
+        service_(BraveVpnServiceFactory::GetForProfile(browser_->profile())) {
+    DCHECK(service_);
+    Observe(service_);
+    Build(service_->is_purchased_user());
+  }
+
+  ~VPNButtonMenuModel() override = default;
+  VPNButtonMenuModel(const VPNButtonMenuModel&) = delete;
+  VPNButtonMenuModel& operator=(const VPNButtonMenuModel&) = delete;
+
+ private:
+  // ui::SimpleMenuModel::Delegate override:
+  void ExecuteCommand(int command_id, int event_flags) override {
+    chrome::ExecuteCommand(browser_, command_id);
+  }
+
+  // BraveVPNServiceObserver overrides:
+  void OnPurchasedStateChanged(
+      brave_vpn::mojom::PurchasedState state) override {
+    // Rebuild menu items based on purchased state change.
+    Build(service_->is_purchased_user());
+  }
+
+  void Build(bool purchased) {
```

Comment:
I was confused with `Build` thought it was related to gn build system. Perhaps we should rename this to `CreateMenu` or `CreateList`

Reply:
`Build` is widely term word in menu model codebase in chromium.
I think all menu model uses `Build`.

Reply:
fair enough